src/html/html-element-reference-scanner.ts

```diff
@@ -0,0 +1,52 @@
+import * as dom5 from 'dom5';
+import {ASTNode} from 'parse5';
+import {ScannedElementReference} from '../model/element-reference';
+import {HtmlVisitor, ParsedHtmlDocument} from './html-document';
+import {HtmlScanner} from './html-scanner';
+
+const isCustomElement = dom5.predicates.hasMatchingTagName(/(.+-)+.+/);
+
+export class HtmlElementReferenceScanner implements HtmlScanner {
```

Comment:
Mind adding docs for this class? Also worth pointing out in the docs that this scanner isn't hooked up by default, but its subclass is.